### ISSUE TITLE
Guava Maven Artifact with wrong classifier (-android instead of -jre) gets selected via dependency instead of user selected one

### ISSUE THREAD

**Issue by vorburger** [bazelbuild/rules_jvm_external#872]
I have a full project which illustrates this problem, it's currently in a private repo but I'm hoping to open it within days, if it helps:

Despite using `"com.google.guava:guava:31.1-jre"` in my `MODULE.bazel`, it seemed to pull an older Android instead of JRE version:

```
$ bazelisk query "@maven//:*" | grep guava
Loading: 0 packages loaded
Loading: 0 packages loaded
@maven//:com_google_guava_failureaccess
@maven//:com_google_guava_failureaccess_1_0_1
@maven//:com_google_guava_failureaccess_1_0_1_extension
@maven//:com_google_guava_failureaccess_jar_sources_1_0_1_extension
@maven//:com_google_guava_guava
@maven//:com_google_guava_guava_30_1_1_android
@maven//:com_google_guava_guava_30_1_1_android_extension
@maven//:com_google_guava_guava_jar_sources_30_1_1_android_extension
@maven//:com_google_guava_listenablefuture
@maven//:com_google_guava_listenablefuture_9999_0_empty_to_avoid_conflict_with_guava
@maven//:com_google_guava_listenablefuture_9999_0_empty_to_avoid_conflict_with_guava_extension
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1-sources.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/guava/30.1.1-android/guava-30.1.1-android-sources.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/guava/30.1.1-android/guava-30.1.1-android.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
```

This seems to somehow be caused by me having also had a `"io.kubernetes:client-java:17.0.1"` in my `MODULE.bazel`. When I upgraded that to the latest `"io.kubernetes:client-java:18.0.0"` then it worked:

```
$ bazelisk query "@maven//:*" | grep guava
@maven//:com_google_guava_failureaccess
@maven//:com_google_guava_failureaccess_1_0_1
@maven//:com_google_guava_failureaccess_1_0_1_extension
@maven//:com_google_guava_failureaccess_jar_sources_1_0_1_extension
@maven//:com_google_guava_guava
@maven//:com_google_guava_guava_31_1_jre
@maven//:com_google_guava_guava_31_1_jre_extension
@maven//:com_google_guava_guava_jar_sources_31_1_jre_extension
@maven//:com_google_guava_listenablefuture
@maven//:com_google_guava_listenablefuture_9999_0_empty_to_avoid_conflict_with_guava
@maven//:com_google_guava_listenablefuture_9999_0_empty_to_avoid_conflict_with_guava_extension
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1-sources.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/failureaccess/1.0.1/failureaccess-1.0.1.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/guava/31.1-jre/guava-31.1-jre-sources.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/guava/31.1-jre/guava-31.1-jre.jar
@maven//:v1/https/repo1.maven.org/maven2/com/google/guava/listenablefuture/9999.0-empty-to-avoid-conflict-with-guava/listenablefuture-9999.0-empty-to-avoid-conflict-with-guava.jar
```

But this seems wrong... as per e.g. #107, and "common expectation" (based on experience with how e.g. Maven & Gradle resolve these kind of conflicting dependencies), it should prioritize resolving via user selected one instead of from a transitive dependency.

There were a few other dependencies, but I don't think those mattered:

```
bazel_dep(name = "rules_jvm_external", version = "4.5")

maven = use_extension("@rules_jvm_external//:extensions.bzl", "maven")
maven.install(
    artifacts = [
        "com.google.guava:guava:31.1-jre",
        "com.google.truth:truth:1.1.3",
        "io.kubernetes:client-java:17.0.1",
        # "io.netty:netty-transport:4.1.89.Final",
        # "io.netty:netty-resolver:4.1.89.Final",
        # "org.apache.tomcat:tomcat-annotations-api:10.1.5",
        "org.slf4j:slf4j-simple:2.0.6",
    ],
    fetch_sources = True,
    lock_file = "//:maven_install.json",
)
use_repo(maven, "maven", "unpinned_maven")
```

**Comment by vorburger** [comment]
> bazel_dep(name = "rules_jvm_external", version = "4.5")

Oh, this is outdated... maybe this is already fixed in current 5.1? I'll give it a try.

PS: There must be something which can automagically update `bazel_dep` to latest versions?

**Comment by vorburger** [comment]
> Oh, this is outdated... maybe this is already fixed in current 5.1? I'll give it a try.

Nope! I can reproduce this even after I bump from 4.5 to 5.1.

This really does seem to be a bug?

I have a full project which illustrates this problem, it's currently in a private repo but I'm hoping to open it within days, if there is interest.

**Comment by vorburger** [comment]
Duh! If you `REPIN=1 bazel run @unpinned_maven//:pin` after bumping the version then it actually is fixed.
